cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/metrics/Clock.java

```diff
@@ -0,0 +1,59 @@
+package io.smartcat.cassandra.diagnostics.metrics;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+
+/**
+ * An abstraction for how time passes.
+ */
+public abstract class Clock {
+    /**
+     * Returns the current time tick.
+     *
+     * @return time tick in nanoseconds
+     */
+    public abstract long getTick();
```

Comment:
I would rather place nanoseconds in method name than in JavaDoc, better tells story